test/services/next_references/next_transaction_file_reference.service.test.js

```diff
@@ -26,43 +26,47 @@ describe('Next Transaction File Reference service', () => {
 
   describe('When a valid region and regime are specified', () => {
     it('returns a correctly formatted file reference', async () => {
-      const result = await NextTransactionFileReferenceService.go(regime, 'R')
+      const result = await NextTransactionFileReferenceService.go(regime, 'R', 'presroc')
 
       expect(result).to.equal('nalri50001')
     })
 
     describe('the file reference generated', () => {
       it('increments with each call', async () => {
-        const result = await NextTransactionFileReferenceService.go(regime, 'R')
-        const secondResult = await NextTransactionFileReferenceService.go(regime, 'R')
+        const result = await NextTransactionFileReferenceService.go(regime, 'R', 'presroc')
+        const secondResult = await NextTransactionFileReferenceService.go(regime, 'R', 'presroc')
 
-        // The call to slice(-1) grabs the last character from the returned string
-        expect(result.slice(-1)).to.equal('1')
-        expect(secondResult.slice(-1)).to.equal('2')
+        expect(result).endsWith('1')
+        expect(secondResult).endsWith('2')
       })
 
-      it('increments with each call independently for each regime & region', async () => {
+      it('increments with each call independently for each regime, region and ruleset', async () => {
```

Comment:
Is this necessarily true and/or clear? The file reference sequence is only linked to regime and region. Based on this statement I'd expect to see a test using `rulesetResult` that also ended with 1 when I just changed the ruleset param.

Perhaps a negative test, for example, `does not increment independently for each ruleset` would demonstrate in the tests the behaviour we expect. 🤷

Reply:
Agreed. I've reworked tests to show that the sequence is unaffected by ruleset -- so we show that the sequence increments for the same regime/region regardless of ruleset, but also that the sequence doesn't increment for a different regime/region, irrespective of whether the ruleset is the same or different. Possibly overkill but we're covering all the bases!